ale_linters/cuda/clang.vim

```diff
@@ -0,0 +1,31 @@
+" Author: blahgeek <i@blahgeek.com>
+" Description: Clang linter for cuda files
+"
+" Almost same as clang linter for cpp
+
+call ale#Set('cuda_clang_executable', 'clang++')
+call ale#Set('cuda_clang_options', '-std=c++14 -Wall')
+
+function! ale_linters#cuda#clang#GetExecutable(buffer) abort
+    return ale#Var(a:buffer, 'cuda_clang_executable')
+endfunction
+
+function! ale_linters#cuda#clang#GetCommand(buffer) abort
+    let l:paths = ale#c#FindLocalHeaderPaths(a:buffer)
+
+    " -iquote with the directory the file is in makes #include work for
+    "  headers in the same directory.
+    return ale#Escape(ale_linters#cuda#clang#GetExecutable(a:buffer))
+    \   . ' -S -x cuda -fsyntax-only '
+    \   . '-iquote ' . ale#Escape(fnamemodify(bufname(a:buffer), ':p:h')) . ' '
```

Comment:
I recommend using `expand('#' . a:buffer . ':p:h')` here instead of `fnamemodify(bufname(a:buffer), ':p:h')`.